core/vm/contracts.go

```diff
@@ -1132,3 +1132,38 @@ func (c *getVerifiedSealBitmap) Run(input []byte, caller common.Address, evm *EV
 
 	return common.LeftPadBytes(extra.AggregatedSeal.Bitmap.Bytes()[:], 32), gas, nil
 }
+
+// cip20HashFunctions is a precompile to compute any of several
+// cryprographic hash functions
+type cip20HashFunctions struct{}
+
+func (c *cip20HashFunctions) RequiredGas(input []byte) uint64 {
+	if len(input) == 0 {
+		return params.InvalidCip20Gas
+	}
+
+	if h, ok := Cip20HashesDonut[input[0]]; ok {
```

Comment:
we have a static reference to `Cip20HashesDonut` this can be easily fixed by adding a field to `struct cip20HashFunctions` which is the hashes map. And so in line 195 you set the donut hashes, for future hardforks we change that in the precompilesForXXX function